lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -14,11 +14,13 @@ def initialize(files, concurrency: DEFAULT_CONCURRENCY)
 
         def run(&block)
           queue = build_queue
+          lock = Mutex.new
 
           @workers = thread_count.times.map do
             Thread.new do
-              while !queue.empty? && (item = queue.pop(true))
-                yield item
+              until queue.empty?
+                item = lock.synchronize { queue.pop(true) unless queue.empty? }
+                yield item if item
```

Comment:
Oh I see. It's not a global lock, you have to use the same instance in both (all) threads for it to actually prevent the concurrency. Sorry about that.

Instance var seems fine. The FP purist in me would probably pass it into `#next_item`.